### DISCUSSION TITLE
Cannot read properties of undefined (reading 'key')

### DISCUSSION THREAD

**Discussion by mrieder** [requarks/wiki#5301] [Error / Bug Report]
When only one authentication provider is active (Hide Local Authentication Provider setting is enabled), it is still possible for a user to disable/de-select the provider by clicking on it on the login page. Then the user gets the error message "_**Cannot read properties of undefined (reading 'key')**_".

Example: 
By default the LDAP/AD provider is active (indicated by a light blue background). Login works as expected.
![image](https://user-images.githubusercontent.com/20661789/168761153-271990f2-b190-476b-9e57-c4b8ed40d036.png)

If the users clicks at the provider, it gets deactivated (indicated by a white background) and the users gets the error message.
![image](https://user-images.githubusercontent.com/20661789/168761273-f9cec21a-25dd-44ff-bfec-cbe846517423.png)

Suggestion: If only one authentication provider is active, it should be impossible to disable it.

**Comment by bendres97** [comment]
+1 on this. I wasted an unholy amount of time trying to figure out why my local login wasn't working. I was clicking local login without realizing it was already selected for me, thus unselecting it and then not being able to log in.

**Comment by vorwd** [comment]
+1... seems strange to allow it.

**Comment by ascheucher** [comment]
Two years later, the same issue is still a time sink...

**Comment by VarIr** [comment]
I would love to see a more helpful error message (something like: "No authentication provider selected. Please choose one of the available providers"). This would also cover the case when multiple providers are available.
Alternatively, disabling de-selecting altogether might be an even better solution.
